src/rattler_build_conda_compat/modify_recipe.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import copy
+import hashlib
+import io
+import re
+import logging
+from typing import TYPE_CHECKING, Any, Literal
+
+import requests
+from ruamel.yaml import YAML
+
+from rattler_build_conda_compat.jinja.jinja import jinja_env, load_recipe_context
+from rattler_build_conda_compat.recipe_sources import Source, get_all_sources
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+yaml = YAML()
```

Comment:
wdyt about extracting YAML loader configuration in one place so it can be used by other loaders?

Reply:
yes, I also want to get rid of any `PyYAML` and just use `ruamel.YAML` everywhere.

Reply:
In fact, that's the next PR that I wanted to send, so let's maybe wait until then?

Reply:
sounds good to me